crates/next-core/src/next_client/transition.rs

```diff
@@ -83,18 +69,30 @@ impl Transition for NextClientTransition {
         asset: AssetVc,
         context: ModuleAssetContextVc,
     ) -> Result<AssetVc> {
-        let chunkable_asset = match ChunkableAssetVc::resolve_from(asset).await? {
-            Some(chunkable_asset) => chunkable_asset,
-            None => bail!(
-                "asset {} is not chunkable",
-                asset.ident().to_string().await?
-            ),
+        let internal_asset = if self.is_app {
+            next_asset("entry/app/hydrate.tsx")
+        } else {
+            next_asset("entry/next-hydrate.tsx")
```

Comment:
We should move `entry/next-hydrate.tsx` to `entry/page/hydrate.tsx` to make it symmetric with `entry/app`.